### PR TITLE
apply.to.all.orgs

### PR DESCRIPTION
**Pull request by deadlyicon**

### CODE REVIEW

File: isValidAccountDataSectionValue.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  case 'personal_data':
+    return typeof value === 'string' || value === null;
+  case 'shared_personal_data':
+  case 'consents':
+    return typeof value === 'boolean';
+  case 'communication_channels':
+    return !!value && typeof value.enabled === 'boolean';
+  default:
+    throw new Error(`invalid account data section: "${section}"`);
+  }
```
Comment (nodatall) [inline isValidAccountDataSectionValue.js]: indent code inside the switch
```
switch(section) {
  case 'personal_data':
    return typeof value === 'string' || value === null;
  case 'shared_personal_data':
  case 'consents':
    return typeof value === 'boolean';
  case 'communication_channels':
    return !!value && typeof value.enabled === 'boolean';
  default:
    throw new Error(`invalid account data section: "${section}"`);
}
```

File: normalizeAccountData.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+const DEFAULT_ACCOUNT_DATA = require('./default_account_data');
+const isValidAccountDataSectionValue = require('./isValidAccountDataSectionValue');
+
+module.exports = function normalizeAccountData(accountData) {
+  const normalizedAccountData = {};
+  for (const section in DEFAULT_ACCOUNT_DATA){
+    normalizedAccountData[section] = {};
+    for (const key in DEFAULT_ACCOUNT_DATA[section]){
+      let newValue = accountData && accountData[section] && accountData[section][key];
```
Comment (nodatall) [inline normalizeAccountData.js]: If there is no accountData shouldn't we handle that at the top of this function?
```
module.exports = function normalizeAccountData(accountData) {
   const normalizedAccountData = {}
   if (!accountData) return normalizedAccountData
...
}
```

File: test/default_account_data.spec.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const DEFAULT_ACCOUNT_DATA = require('../default_account_data');
+
+describe('default_account_data', function(){
+  it('should be as expected', function(){
+    expect(DEFAULT_ACCOUNT_DATA).to.deep.equal({
+      shared_personal_data: {
```
Comment (nodatall) [inline test/default_account_data.spec.js]: Do we really want all `shared_personal_data` values to default to `true`?

File: test/scenarios/managing_account_data.spec.js [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+          'Brand Marketing': true,
+          'Product Marketing': true,
+        },
+        communication_channels: {
+          email_media: true,
+          fax_media: true,
+        },
+      };
+      organizationAccountData = stripNonRequestedAccountData(
+        mergeAccountData(DEFAULT_ACCOUNT_DATA),
```
Comment (nodatall) [inline test/scenarios/managing_account_data.spec.js]: is mergeAccountData necessary here?

File: mergeAccountData.js [hunk 5]
```diff
@@ -1,42 +1,32 @@
...
+      }
     }
-  });
-
+    if (Object.keys(mergedAccountData[section]).length === 0)
+      delete mergedAccountData[section];
+  }
   return mergedAccountData;
-};
+};
```
Comment (nodatall) [inline mergeAccountData.js]: add newline

File: normalizeAccountData.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+const defaultAccountDataSectionValue = function(section){
+  switch(section) {
+  case 'shared_personal_data':
+  case 'consents':
+    return false;
+  case 'communication_channels':
+    return { enabled: false };
+  case 'personal_data':
+    return '';
+  }
```
Comment (nodatall) [inline normalizeAccountData.js]: indent code inside switch?

File: stripNonRequestedAccountData.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    for (const key in strippedAccountData[section]){
+      if (requestedData[requestedDataKey][key] !== true)
+        delete strippedAccountData[section][key];
+    }
+    if (Object.keys(strippedAccountData[section]).length === 0)
+      delete strippedAccountData[section];
+  }
+  // console.log('strippedAccountData', strippedAccountData)
```
Comment (nodatall) [inline stripNonRequestedAccountData.js]: remove comment

File: test/scenarios/applying_an_organizations_account_data_to_default_account_data.spec.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+context('applying an organizations account data to default account data', function(){
+  it('should have tests');
+  context('when applying a change to a field that is not requested by the organization', function(){
+    it('should ignore the change');
+  });
+});
```
Comment (nodatall) [inline test/scenarios/applying_an_organizations_account_data_to_default_account_data.spec.js]: newline

File: test/scenarios/applying_default_account_data_to_an_organizations_account_data.spec.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+context('applying default account data to an organizations account data', function(){
+  it('should have tests');
+  context('when applying a change to a field that is not requested by the organization', function(){
+    it('should ignore the change');
+  });
+});
```
Comment (nodatall) [inline test/scenarios/applying_default_account_data_to_an_organizations_account_data.spec.js]: newline

File: test/scenarios/staging_default_account_data_changes.spec.js [hunk 10]
```diff
@@ -0,0 +1,226 @@
+'use strict';
+
+// const deepfreeze = require('deep-freeze-node');
```
Comment (nodatall) [inline test/scenarios/staging_default_account_data_changes.spec.js]: remove comment

File: isValidAccountDataSectionValue.js (hunk 1 above)
Comment (deadlyicon) [reply isValidAccountDataSectionValue.js]: not according to the linter

File: normalizeAccountData.js (hunk 2 above)
Comment (deadlyicon) [reply normalizeAccountData.js]: nope, its okay if you give it nothing

File: normalizeAccountData.js (hunk 6 above)
Comment (deadlyicon) [reply normalizeAccountData.js]: not according to the linter

File: test/scenarios/staging_organization_account_data_changes.spec.js [hunk 11]
```diff
@@ -0,0 +1,246 @@
+'use strict';
+
+// const deepfreeze = require('deep-freeze-node');
```
Comment (nodatall) [inline test/scenarios/staging_organization_account_data_changes.spec.js]: remove comment

File: test/stageAccountDataChanges.spec.js [hunk 12]
```diff
@@ -0,0 +1,246 @@
+'use strict';
+
+const deepfreeze = require('deep-freeze-node');
+const PERSONAL_DATA_KEYS = require('../personal_data_keys');
+const CONSENTS = require('../consents');
+const COMMUNICATION_CHANNELS = require('../communication_channels');
+// const normalizeAccountData = require('../normalizeAccountData');
```
Comment (nodatall) [inline test/stageAccountDataChanges.spec.js]: comment

**Pull request merged**